src/test/groovy/com/palantir/gradle/ConfigurationCacheTest.groovy

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle
+
+import nebula.test.IntegrationTestKitSpec
+
+class ConfigurationCacheTest extends IntegrationTestKitSpec {
+
+    def setup() {
+        // language=Gradle
+        buildFile << """
+            apply plugin: 'com.palantir.git-version'
+            apply plugin: 'java-library'
+
+            repositories {
+                mavenCentral()
+                mavenLocal()
+            }
+
+            version gitVersion()
```

Comment:
if we have to exercise the configuration cache inputs via these calls, then we should also do it for the other apis `gradle-git-version` offers.

That would likely mean move this out of `setup` and into specific test methods for each piece we offer